src/command/expire/expire.c

```diff
@@ -900,68 +899,103 @@ cmdExpire(void)
+                    cfgOptionIdxStrNull(cfgOptRepoCipherPass, repoIdx));
+
+                // If a backupLabel was set, then attempt to expire the requested backup
+                if (adhocBackupLabel != NULL)
                 {
-                    adhocBackupFound = true;
-                    expireAdhocBackup(infoBackup, adhocBackupLabel, repoIdx);
-                }
+                    if (infoBackupDataByLabel(infoBackup, adhocBackupLabel) != NULL)
+                    {
+                        adhocBackupFound = true;
+                        expireAdhocBackup(infoBackup, adhocBackupLabel, repoIdx);
+                    }
 
-                // If the adhoc backup was not found and this was the last repo to check, then log a warning but continue to process
-                // the expiration based on retention
-                if (!adhocBackupFound && repoIdx == repoIdxMax)
+                    // If the adhoc backup was not found and this was the last repo to check, then log a warning but continue to process
+                    // the expiration based on retention
+                    if (!adhocBackupFound && repoIdx == repoIdxMax)
+                    {
+                        LOG_WARN_FMT(
+                            "backup %s does not exist\nHINT: run the info command and confirm the backup is listed",
+                            strZ(adhocBackupLabel));
+                    }
+                }
+                else
                 {
-                    LOG_WARN_FMT(
-                        "backup %s does not exist\nHINT: run the info command and confirm the backup is listed",
-                        strZ(adhocBackupLabel));
+                    // If time-based retention for full backups is set, then expire based on time period
+                    if (timeBasedFullRetention)
+                    {
+                        // If a time period was provided then run time-based expiration otherwise do nothing (the user has already been
+                        // warned by the config system that retention-full was not set)
+                        if (cfgOptionIdxTest(cfgOptRepoRetentionFull, repoIdx))
+                        {
+                            expireTimeBasedBackup(
+                                infoBackup, time(NULL) - (time_t)(cfgOptionUInt(cfgOptRepoRetentionFull) * SEC_PER_DAY), repoIdx);
+                        }
+                    }
+                    else
+                        expireFullBackup(infoBackup, repoIdx);
+
+                    expireDiffBackup(infoBackup, repoIdx);
                 }
             }
-            else
+            CATCH_ANY()
+            {
+                LOG_ERROR_FMT(
+                    errorTypeCode(errorType()), "repo%u: %s", cfgOptionGroupIdxToKey(cfgOptGrpRepo, repoIdx), errorMessage());
+                repoError = true;
+                errorTotal++;
+            }
+            TRY_END();
+
+            if (!repoError)
```

Comment:
I can't figure out why expire is broken into two try blocks. Why not just run everything in the first try block?

Reply:
Fixed https://github.com/pgbackrest/pgbackrest/pull/1308/commits/54e3bf08da92f1c157dc6d3f79f80439e4f35265